ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,239 @@
+        match alternate_handling {
+            ffi::ICU4XCollatorAlternateHandling::Auto => None,
+            ffi::ICU4XCollatorAlternateHandling::NonIgnorable => {
+                Some(AlternateHandling::NonIgnorable)
+            }
+            ffi::ICU4XCollatorAlternateHandling::Shifted => Some(AlternateHandling::Shifted),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorCaseFirst> for Option<CaseFirst> {
+    fn from(case_first: ffi::ICU4XCollatorCaseFirst) -> Option<CaseFirst> {
+        match case_first {
+            ffi::ICU4XCollatorCaseFirst::Auto => None,
+            ffi::ICU4XCollatorCaseFirst::Off => Some(CaseFirst::Off),
+            ffi::ICU4XCollatorCaseFirst::LowerFirst => Some(CaseFirst::LowerFirst),
+            ffi::ICU4XCollatorCaseFirst::UpperFirst => Some(CaseFirst::UpperFirst),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorMaxVariable> for Option<MaxVariable> {
+    fn from(max_variable: ffi::ICU4XCollatorMaxVariable) -> Option<MaxVariable> {
+        match max_variable {
+            ffi::ICU4XCollatorMaxVariable::Auto => None,
+            ffi::ICU4XCollatorMaxVariable::Space => Some(MaxVariable::Space),
+            ffi::ICU4XCollatorMaxVariable::Punctuation => Some(MaxVariable::Punctuation),
+            ffi::ICU4XCollatorMaxVariable::Symbol => Some(MaxVariable::Symbol),
+            ffi::ICU4XCollatorMaxVariable::Currency => Some(MaxVariable::Currency),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorCaseLevel> for Option<CaseLevel> {
+    fn from(case_level: ffi::ICU4XCollatorCaseLevel) -> Option<CaseLevel> {
+        match case_level {
+            ffi::ICU4XCollatorCaseLevel::Auto => None,
+            ffi::ICU4XCollatorCaseLevel::Off => Some(CaseLevel::Off),
+            ffi::ICU4XCollatorCaseLevel::On => Some(CaseLevel::On),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorNumeric> for Option<Numeric> {
+    fn from(numeric: ffi::ICU4XCollatorNumeric) -> Option<Numeric> {
+        match numeric {
+            ffi::ICU4XCollatorNumeric::Auto => None,
+            ffi::ICU4XCollatorNumeric::Off => Some(Numeric::Off),
+            ffi::ICU4XCollatorNumeric::On => Some(Numeric::On),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorBackwardSecondLevel> for Option<BackwardSecondLevel> {
+    fn from(
+        backward_second_level: ffi::ICU4XCollatorBackwardSecondLevel,
+    ) -> Option<BackwardSecondLevel> {
+        match backward_second_level {
+            ffi::ICU4XCollatorBackwardSecondLevel::Auto => None,
+            ffi::ICU4XCollatorBackwardSecondLevel::Off => Some(BackwardSecondLevel::Off),
+            ffi::ICU4XCollatorBackwardSecondLevel::On => Some(BackwardSecondLevel::On),
+        }
+    }
+}
+
+impl From<ffi::ICU4XCollatorOptions> for CollatorOptions {
+    fn from(options: ffi::ICU4XCollatorOptions) -> CollatorOptions {
+        let mut result = CollatorOptions::new();
+        result.strength = Option::<Strength>::from(options.strength);
```

Comment:
Nit (optional): I think this would be clearer and more readable if you used `options.strength.into()`

Reply:
Done. I prefer `From:: ...` because I am very wary of `.into()` because I think it is often used for non-simple conversions (type A -> type B where A and B are very different, and the conversion is non-trivial).  It makes code quite unreadable.

But I accept this to be the first instance in a list of begrudging exceptions -- where you are copying over a field from one struct to another as-is.

Although I still do prefer the `From...` syntax since it looks like a function rather than a method, and that better communicates the stateless nature of the conversion.